backend/src/test/java/com/ody/route/service/RouteServiceTest.java

```diff
@@ -0,0 +1,72 @@
+package com.ody.route.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import com.ody.common.BaseServiceTest;
+import com.ody.meeting.domain.Coordinates;
+import com.ody.route.domain.RouteTime;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+class RouteServiceTest extends BaseServiceTest {
+
+    @MockBean
+    @Qualifier("odsay")
+    private RouteClient odsayRouteClient;
+
+    @MockBean
+    @Qualifier("google")
+    private RouteClient googleRouteClient;
+
+    @Autowired
+    private RouteService routeService;
+
+    @DisplayName("OdsayRouteClient에 에러가 발생하지 않으면 첫 번째 외부 API를 사용해 소요시간을 반환한다.")
+    @Test
+    void calculateRouteTimeByOdsayRouteClient() {
+        Coordinates origin = new Coordinates("37.505419", "127.050817");
+        Coordinates target = new Coordinates("37.515253", "127.102895");
+
+        Mockito.when(odsayRouteClient.calculateRouteTime(origin, target))
+                .thenReturn(new RouteTime(16));
+
+        Mockito.when(googleRouteClient.calculateRouteTime(origin, target))
+                .thenReturn(new RouteTime(18));
+
+        long result = routeService.calculateRouteTime(origin, target).getMinutes();
+        RouteTime expectRouteTime = new RouteTime(16);
+
+        assertAll(
+                () -> assertThat(result).isEqualTo(expectRouteTime.getMinutes()),
+                () -> Mockito.verify(odsayRouteClient, Mockito.times(1)).calculateRouteTime(origin, target),
+                () -> Mockito.verify(googleRouteClient, Mockito.times(0)).calculateRouteTime(origin, target)
+        );
+    }
+
+    @DisplayName("OdsayRouteClient에 에러가 발생하면 그 다음 요소인 Google API를 사용해 소요시간을 반환한다.")
+    @Test
+    void calculateRouteTimeByGoogleRouteClient() {
+        Coordinates origin = new Coordinates("37.505419", "127.050817");
+        Coordinates target = new Coordinates("37.515253", "127.102895");
+
+        Mockito.when(odsayRouteClient.calculateRouteTime(origin, target))
```

Comment:
mock 연속적인 동작에 대해 체이닝을 통해 지정이 가능해요.
```suggestion
        Mockito.when(odsayRouteClient.calculateRouteTime(origin, target))
                      .thenThrow() //1번째 호출 동작
                      .thenReturn() //2번째 호출 동작
       
```

Reply:
`odsayRouteClient.calculateRouteTime()`를 호출 했을 때  예외를 던지고
`googleRouteClient.calculateRouteTime()`을 호출 했을 때는 결과를 반환하게 해
`routeService.calculateRouteTime()`을 실행했을 때 예외가 발생한 `odsayRouteClient`는 패스하고 `googleRouteClient`를 통해 소요시간을 계산하는지 확인하는 테스트에요

위 처럼 연속적으로 호출하도록 할 상황은 아닌데 콜리의 의도는 무엇이였을까요 ?? 🤔

Reply:
> 위 처럼 연속적으로 호출하도록 할 상황은 아닌데 콜리의 의도는 무엇이였을까요 ?? 🤔

아 다른 mockito의 stubbing이었군요! 꼼꼼히 리뷰해보겠습니다 😭